### PR TITLE
Update shouty.js to bring inline with shouty.java.  

### PR DESCRIPTION
**Pull request by temyers**
Updated the shouty.js codebase to be consistent with the current shouty.java implementation

* Added Jest unit test framework, with failing unit test
* Added implementation in line with shouty.java
* Added sample scenarios, with Step Definition implementation

### CODE REVIEW

**Comment by temyers** [comment]
The changes here are now complete and ready for review.

File: .gitignore [hunk 1]
```diff
@@ -1 +1,2 @@
 node_modules
+jest/
```
Comment (jbpros) [inline .gitignore]: Can be removed, right?

File: features/step_definitions/shout_steps.js [hunk 2]
```diff
@@ -1,7 +1,35 @@
 var assert = require('assert');
-var Shouty = require('../../lib/shouty');
+  var Shouty = require('../../lib/shouty');
```
Comment (jbpros) [inline features/step_definitions/shout_steps.js]: Weird indentation here :)

**Review by jbpros — CHANGES_REQUESTED** [review]
This is great!

I've discussed briefly with @tooky, we were wondering if we should use ES6 `class` and other goodies. I think we should, so that we teach BDD through modern JS :)

It can be done in a separate PR though.

I'm curious: why did you switch from jest to mocha @temyers?

**Comment by temyers** [comment]
I switched from Jest to Mocha because Jest was taking far too long to execute the tests.  ~5s, compared to a few < 50ms with Mocha.

File: .gitignore (hunk 1 above)
Comment (temyers) [reply .gitignore]: 👍

File: features/step_definitions/shout_steps.js (hunk 2 above)
Comment (temyers) [reply features/step_definitions/shout_steps.js]: 👍

**Comment by jbpros** [comment]
Thanks @temyers!

**Pull request merged**